client/src/components/App/app.jsx

```diff
@@ -60,13 +62,28 @@ class App extends React.Component {
 
   toggleModal(){
     console.log('clicked');
-    this.setState(prevState =>({
+    // this.setState(prevState =>({
+    //   modalOpen: !prevState.modalOpen
+    // }),()=>{
+    //   console.log(this.state.modalOpen);
+    // })
+
+    if (!this.state.modalOpen) {
+      document.addEventListener("click", this.handleOutsideClick, false);
+    } else {
+      document.removeEventListener("click", this.handleOutsideClick, false);
+    }
+
+    this.setState(prevState => ({
       modalOpen: !prevState.modalOpen
-    }),()=>{
-      console.log(this.state.modalOpen);
-    })
+    }));
   };
 
+  handleOutsideClick(e){
+
+    if (!this.node.contains(e.target) && e.target.title==="insidemodal") this.toggleModal();
```

Comment:
Interesting to see the similarities and differences in our approaches to toggling our modals. addEventListener and removeEventListener were two methods I had forgotten about until I was working thru this problem

Reply:
Yeah, I didnt want to go through rewriting everything using hooks loll.

Reply:
😆 i feel that